src/preamble.js

```diff
@@ -2221,6 +2221,19 @@ function integrateWasmJS(Module) {
       Module['asm'] = exports;
       Module["usingWasm"] = true;
     }
+
+    // User shell pages can write their own Module.instantiateWasm = function(imports, successCallback) callback
+    // to manually instantiate the Wasm module themselves. This allows pages to run the instantiation parallel
+    // to any other async startup actions they are performing.
+    if (Module['instantiateWasm']) {
+      try {
+        return Module['instantiateWasm'](info, receiveInstance);
```

Comment:
should this create a run dependency, like the async code below does? the docs say async is supported. if the user does async without a run dep, it can fail depending on timing if they don't manually add a run dep.